src/scripts/extension-utils.js

```diff
@@ -72,10 +70,11 @@ const sendAsyncMessageToContentScript = async payload => {
   return response
 }
 
-export const getAccounts = () => {
-  sendMessageToContentScript({
+export const getWalletsFromExtension = async () => {
+  const wallets = await sendAsyncMessageToContentScript({
```

Comment:
Do we need my `sendAsyncMessageToContentScript` with the new async handling in the extension?

Reply:
We can still use it. We just don't need to 'skipResponse'.

Ideally we could handle every response in `processMessage((store, type, payload)`. It means all responses go in the store. Would that work for every request? Sign requests?

Reply:
Maybe. But it would mean that we need to handle the pending signRequest state in the store. Might be reasonable.

Reply:
Technically we should store all state in the store. But, it's hard to tell if this (the actionmodal) is an exception)

…thinking. 🤔